request/request.go

```diff
@@ -0,0 +1,68 @@
+package request
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+type Record struct {
+	cost  time.Duration
+	code  string
+	error string
+	bytes int
+}
+
+func newRecord(resp *http.Response, t time.Duration) Record {
+	var code string
+
+	switch resp.StatusCode / 100 {
+	case 1:
+		code = "1xx"
+	case 2:
+		code = "2xx"
+	case 3:
+		code = "3xx"
+	case 4:
+		code = "4xx"
+	case 5:
+		code = "5xx"
+	}
+
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+
+	r := Record{
+		code:  code,
+		cost:  t,
+		bytes: len(body),
+	}
+
+	if err != nil {
+		r.error = fmt.Sprint(err)
+	}
+
+	return r
+}
+
+func (r Record) String() string {
+	return fmt.Sprintf("status %s, took %s", r.code, r.cost)
+}
+
+func Do(url string, timeout time.Duration) Record {
+	client := http.Client{
+		// Timeout includes connection time, any redirects, and reading the response body.
+		// We may want exclude reading the response body in our benchmark tool.
```

Comment:
See this [answer](https://stackoverflow.com/a/16930649) for "How to set timeout for http.Get() requests in Golang?".

Reply:
This might become its own issue for later ?

Reply:
I'm wondering whether we should include it or not, it could be a valid indicator of an endpoint's performance: if it's super quick but delivers a super long body to read we could miss that information 🤔 Could also be an option.

Worth discussing indeed!

Reply:
Yes it can go both ways. This need more attention and discussion.
Let's not bother with it for now, open an issue and tag it for consideration? So we don't block this PR on that.